### PR TITLE
[move][std] Add `std::uq64_64`

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- Adds `uq64_64` module, which is based on `uq32_32`
- Revert snapshot for protocol config `70` (reverting #20531)

## Test plan 

- New tests

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [X] CLI: New Move module `std::uq64_64` for larger-precision, fixed-point numbers
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-framework/packages/move-stdlib/sources/macros.move [hunk 1]
```diff
@@ -145,3 +145,93 @@ public macro fun try_as_u128($x: _): Option<u128> {
...
+    $abort_denominator: _,
+    $abort_quotient_too_small: _,
+    $abort_quotient_too_large: _,
+): $T {
+    let numerator = $numerator;
+    let denominator = $denominator;
+    if (denominator == 0) $abort_denominator;
+
+    // Scale the numerator to have 64 fractional bits and the denominator to have 32 fractional
+    // bits, so that the quotient will have 32 fractional bits.
```
Comment (karlwuest) [inline crates/sui-framework/packages/move-stdlib/sources/macros.move:164]: Nit: The comment should probably use $t_bits and $fractional_bits.

File: crates/sui-framework/packages/move-stdlib/sources/macros.move [hunk 2]
```diff
@@ -145,3 +145,93 @@ public macro fun try_as_u128($x: _): Option<u128> {
...
+    let scaled_numerator = numerator as $U << $t_bits;
+    let scaled_denominator = denominator as $U << $fractional_bits;
```
Comment (karlwuest) [inline crates/sui-framework/packages/move-stdlib/sources/macros.move:166]: It doesn't matter for the two concrete instantiations currently, but I think this doesn't work correctly if t_bits is not 2*fractional_bits. This should probably be:

```suggestion
    let scaled_numerator = numerator as $U << $t_bits;
    let scaled_denominator = denominator as $U << ($t_bits - $fractional_bits);
```

File: crates/sui-framework/packages/move-stdlib/sources/macros.move [hunk 3]
```diff
@@ -145,3 +145,93 @@ public macro fun try_as_u128($x: _): Option<u128> {
...
+    if (quotient == 0 && numerator != 0) $abort_quotient_too_small;
+
+    // Return the quotient as a fixed-point number. We first need to check whether the cast
+    // can succeed.
+    if (quotient > $max_t as $U) $abort_quotient_too_large;
+    quotient as $T
+}
+
+public macro fun uq_from_int<$T, $U>($integer: $T, $t_bits: u8): $U {
+    ($integer as $U) << $t_bits
```
Comment (karlwuest) [inline crates/sui-framework/packages/move-stdlib/sources/macros.move:179]: Nit:
```suggestion
public macro fun uq_from_int<$T, $U>($integer: $T, $fractional_bits: u8): $U {
    ($integer as $U) << $fractional_bits
```

File: crates/sui-framework/packages/move-stdlib/sources/macros.move [hunk 4]
```diff
@@ -145,3 +145,93 @@ public macro fun try_as_u128($x: _): Option<u128> {
...
+    $multiplier: $T,
+    $max_t: $T,
+    $fractional_bits: u8,
+    $abort_overflow: _,
+): $T {
+    // The product of two $T bit values has $U bits, so perform the
+    // multiplication with $U types and keep the full $U bit product
+    // to avoid losing accuracy.
+    let unscaled_product = $val as $U * ($multiplier as $U);
+    // The unscaled product has fractional_bitsfractional bits (from the multiplier)
```
Comment (karlwuest) [inline crates/sui-framework/packages/move-stdlib/sources/macros.move:210]: Nit
```suggestion
    // The unscaled product has fractional_bits fractional bits (from the multiplier)
```

**Review by jonas-lj — APPROVED** [review]
Looks good! The macros are very cool. I've just added some suggestions for the comments and docs, but everything else looks fine.

**Review by cgswords — APPROVED** [review]
LGTM!

**Pull request merged**